pkg/kubelet/kubelet_node_status.go

```diff
@@ -1019,10 +1016,15 @@ func (kl *Kubelet) setNodeOODCondition(node *v1.Node) {
 
 // Maintains Node.Spec.Unschedulable value from previous run of tryUpdateNodeStatus()
 // TODO: why is this a package var?
-var oldNodeUnschedulable bool
+var (
+	oldNodeUnschedulable     bool
+	oldNodeUnschedulableLock sync.Mutex
+)
 
 // record if node schedulable change.
 func (kl *Kubelet) recordNodeSchedulableEvent(node *v1.Node) {
```

Comment:
I don't think `recordNodeScheduableEvent` should be one of `NodeStatusFuncs`.
I'm fine with adding a lock here for now. But actually we should move `recordNodeScheduableEvent` out of `NodeStatusFuncs` and put it where it should be.

@dashpole WDYT?